app/resources/views.py

```diff
@@ -12,51 +13,105 @@ def index():
     return render_template('resources/index.html')
 
 
-@resources.route('/add', methods=['GET', 'POST'])
+@resources.route('/create', methods=['GET', 'POST'])
 @login_required
-def add():
+def create_resource():
     form = ResourceForm()
     if form.validate_on_submit():
-        # Based on form's zip code, load or create ZIPCode and add to db.
+        # Converting Google Places API names to our model names.
+        name = form.name.data
+        street_address = str(form.street_number.data) + ' ' + form.route.data
+        city = form.locality.data
+        state = form.administrative_area_level_1.data
         zip_code = ZIPCode.create_zip_code(form.postal_code.data)
-        # Based on form's address, zip's id, load or create Address, add to db.
-        street_num_route = str(form.street_number.data) + ' ' + form.route.data
-        # TODO: Create helper method in models/location.py.
-        address = Address.query.filter_by(
-            name=form.name.data,
-            street_address=street_num_route,
-            city=form.locality.data,
-            state=form.administrative_area_level_1.data,
-            zip_code_id=zip_code.id).first()
-        if address is None:
-            address = Address(name=form.name.data,
-                              street_address=street_num_route,
-                              city=form.locality.data,
-                              state=form.administrative_area_level_1.data,
-                              zip_code_id=zip_code.id)
-            db.session.add(address)
-            db.session.commit()
-        # Based on form and address id, create a new resource.
-        # TODO: Create helper method in models/resource.py.
-        resource = Resource(name=form.name.data,
-                            description=form.description.data,
-                            website=form.website.data,
-                            address_id=address.id,
-                            user_id=int(current_user.get_id()))
-        db.session.add(resource)
-        db.session.commit()
-        return redirect(url_for('resources.index'))
-    return render_template('resources/add_resource.html', form=form)
+        address = Address.create_address(name,
+                                         street_address,
+                                         city,
+                                         state)
+        address.zip_code = zip_code
+        description = form.description.data
+        website = form.website.data
+        resource = Resource.create_resource(name,
+                                            description,
+                                            website)
+        resource.address = address
+        resource.user = User.query.get(current_user.id)
```

Comment:
let's remove the jank and go with `resource.user_id = current_user.id` or `resource.user = current_user._get_current_object()`, your choice sir